File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.java
class Solution {
    private int[] res; 
    public int[] productExceptSelf(int[] nums) {
        // brute force solution first
        res = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            productResults(nums, i);
        }

        return res;
    }

    public void productResults(int[] nums, int index) {

        int j = index; // j becomes the index we want to skip 
       

        int product = 1; 
        for (int i = 0; i < nums.length; i++) {
            if (i != j) {
                product *= nums[i];
                
            }

        }

        res[j] = product; 
    }
}  

File: Data Structures & Algorithms/longest-palindrome/submission-2.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        
        my_map = {}
        res = 0 
        for c in s: 
            if c not in my_map: 
                my_map[c] = 1 
            else:
                my_map[c] += 1
            
            if my_map[c] % 2 == 0: 
                res += 2
        
        print(my_map)





        for cnt in my_map.values():
            if cnt % 2: 
                res += 1 
                break 
        
        return res